packages/web-twig/src/Resources/components/TextFieldBase/TextFieldBase.twig

```diff
@@ -67,18 +73,50 @@
             {{ _value | raw }}
         </textarea>
     {% else %}
-        <input
-            {{ inputProps(props) }}
-            type="{{ _type }}"
-            id="{{ _id }}"
-            name="{{ _name }}"
-            class="{{ _inputClassName }}"
-            {{ _disabledAttr | raw }}
-            {{ _inputWidthAttr | raw }}
-            {{ _placeholderAttr | raw }}
-            {{ _requiredAttr | raw }}
-            {{ _valueAttr | raw }}
-        />
+        {% if _hasPasswordToggle %}
+            <div class="TextField__passwordToggle">
+                <input
+                    {{ inputProps(props) }}
+                    type="password"
+                    id="{{ _id }}"
+                    name="{{ _name }}"
+                    class="{{ _inputClassName }}"
+                    {{ _disabledAttr | raw }}
+                    {{ _inputWidthAttr | raw }}
+                    {{ _placeholderAttr | raw }}
+                    {{ _requiredAttr | raw }}
+                    {{ _valueAttr | raw }}
+                />
+                <button
+                    type="button"
+                    class="{{ _passwordToggleButtonClassName }}"
+                    role="switch"
+                    aria-pressed="false"
+                    aria-label="Show password"
+                    data-toggle="password"
+                >
+                    <span class="{{ _passwordToggleIconClassName }} {{ _passwordToggleIconHiddenClassName }}">
```

Comment:
Just an idea: how about using our new helper classes `accessibility-open` a `accessibility-closed` also for this context? (They would need an adjustment, of course.)

Reply:
How would we do this then?
```css
.TextField__passwordToggle__button[aria-pressed='true'] > .TextField__passwordToggle__icon--shown {}
```

Reply:
This is what I mean:

```scss
// _accessibility.scss

[aria-pressed="false"] .accessibility-selected,
[aria-pressed="true"] .accessibility-unselected {
  display: none;
}

[aria-pressed="true"] .accessibility-selected,
[aria-pressed="false"] .accessibility-unselected {
  display: initial;
}
```

```suggestion
                    <span class="{{ _passwordToggleIconClassName }} accessibility-selected">
```

Let's not dig into it here, we can talk about it later 🙂.

Reply:
🤔 this might be a good start as I noticed we already have this pattern of hidden text inside icon buttons in Jobs. So we might be able to switch to this and make it more accessible. I added it to our discussion forum and we might create an issue for this later.